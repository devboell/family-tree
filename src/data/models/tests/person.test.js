import knex from '../../connectors'
import PersonModel from '../person'

describe('PersonModel', () => {
  beforeEach(async () => {
    await knex.migrate.rollback()
    await knex.migrate.latest()
    await knex.seed.run()
  })

  afterEach(async () => {
    await knex.migrate.rollback()
  })

  afterAll(() => {
    knex.destroy()
  })

  it('findAll', async () => {
    const persons = await PersonModel.findAll()
    expect(persons).toMatchSnapshot()
  })

  it('findById', async () => {
    const person = await PersonModel.findById(1)
    expect(person).toMatchSnapshot()
  })

  it('findByIds', async () => {
    const person = await PersonModel.findByIds([1, 2, 3])
    expect(person).toMatchSnapshot()
  })

  it('create', async () => {
    const person = await PersonModel.create({ name: 'Joe' })
    expect(person).toMatchSnapshot()
  })

  it('update', async () => {
    const person = await PersonModel.update({ id: 1, name: 'Joe' })
    expect(person).toMatchSnapshot()
  })
})
