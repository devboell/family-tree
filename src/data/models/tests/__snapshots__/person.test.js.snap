// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonModel create 1`] = `
Object {
  "birthDate": null,
  "bornToId": null,
  "deathDate": null,
  "gender": null,
  "id": 18,
  "isMonarch": null,
  "name": "Joe",
  "title": null,
}
`;

exports[`PersonModel findAll 1`] = `
Array [
  Object {
    "birthDate": 1028,
    "bornToId": null,
    "deathDate": 1087,
    "gender": "M",
    "id": 1,
    "isMonarch": 1,
    "name": "William I",
    "title": "King of England",
  },
  Object {
    "birthDate": 1031,
    "bornToId": null,
    "deathDate": 1083,
    "gender": "F",
    "id": 2,
    "isMonarch": 0,
    "name": "Matilde of Flanders",
    "title": "Queen of England",
  },
  Object {
    "birthDate": 1056,
    "bornToId": 1,
    "deathDate": 1100,
    "gender": "M",
    "id": 3,
    "isMonarch": 1,
    "name": "William Rufus",
    "title": "King of England",
  },
  Object {
    "birthDate": 1068,
    "bornToId": 1,
    "deathDate": 1135,
    "gender": "M",
    "id": 4,
    "isMonarch": 1,
    "name": "Henry I",
    "title": "King of England",
  },
  Object {
    "birthDate": 1080,
    "bornToId": null,
    "deathDate": 1118,
    "gender": "F",
    "id": 5,
    "isMonarch": 0,
    "name": "Matilde of Scotland",
    "title": "Queen of England",
  },
  Object {
    "birthDate": 1103,
    "bornToId": 2,
    "deathDate": 1120,
    "gender": "M",
    "id": 6,
    "isMonarch": 0,
    "name": "William Adelin",
    "title": "Duke of Normandy",
  },
  Object {
    "birthDate": 1102,
    "bornToId": 2,
    "deathDate": 1167,
    "gender": "F",
    "id": 7,
    "isMonarch": 0,
    "name": "Empress Matilde",
    "title": "Holy Roman Empress",
  },
  Object {
    "birthDate": 1113,
    "bornToId": null,
    "deathDate": 1151,
    "gender": "M",
    "id": 8,
    "isMonarch": 0,
    "name": "Geoffrey Plantagenet",
    "title": "Count of Anjou",
  },
  Object {
    "birthDate": 1133,
    "bornToId": 3,
    "deathDate": 1189,
    "gender": "M",
    "id": 9,
    "isMonarch": 1,
    "name": "Henry II",
    "title": "King of England",
  },
  Object {
    "birthDate": 1122,
    "bornToId": null,
    "deathDate": 1204,
    "gender": "F",
    "id": 10,
    "isMonarch": 0,
    "name": "Eleanor of Aquitane",
    "title": "Queen of England",
  },
  Object {
    "birthDate": 1157,
    "bornToId": 4,
    "deathDate": 1199,
    "gender": "M",
    "id": 11,
    "isMonarch": 1,
    "name": "Richard I",
    "title": "King of England",
  },
  Object {
    "birthDate": 1166,
    "bornToId": 4,
    "deathDate": 1216,
    "gender": "M",
    "id": 12,
    "isMonarch": 1,
    "name": "John I",
    "title": "King of England",
  },
  Object {
    "birthDate": 1051,
    "bornToId": 1,
    "deathDate": 1134,
    "gender": "M",
    "id": 13,
    "isMonarch": 1,
    "name": "Robert Curthose",
    "title": "Duke of Normandy",
  },
  Object {
    "birthDate": 1103,
    "bornToId": null,
    "deathDate": 1151,
    "gender": "F",
    "id": 14,
    "isMonarch": 0,
    "name": "Adeliza of Louvain",
    "title": "Queen of England",
  },
  Object {
    "birthDate": 1067,
    "bornToId": 1,
    "deathDate": 1137,
    "gender": "F",
    "id": 15,
    "isMonarch": 0,
    "name": "Adela of Normandy",
    "title": "Countess of Blois",
  },
  Object {
    "birthDate": 1045,
    "bornToId": null,
    "deathDate": 1102,
    "gender": "M",
    "id": 16,
    "isMonarch": 0,
    "name": "Stephen II Henry",
    "title": "Count of Blois",
  },
  Object {
    "birthDate": 1092,
    "bornToId": 6,
    "deathDate": 1154,
    "gender": "M",
    "id": 17,
    "isMonarch": 1,
    "name": "Stephen of Blois",
    "title": "King of England",
  },
]
`;

exports[`PersonModel findById 1`] = `
Object {
  "birthDate": 1028,
  "bornToId": null,
  "deathDate": 1087,
  "gender": "M",
  "id": 1,
  "isMonarch": 1,
  "name": "William I",
  "title": "King of England",
}
`;

exports[`PersonModel findByIds 1`] = `
Array [
  Object {
    "birthDate": 1028,
    "bornToId": null,
    "deathDate": 1087,
    "gender": "M",
    "id": 1,
    "isMonarch": 1,
    "name": "William I",
    "title": "King of England",
  },
  Object {
    "birthDate": 1031,
    "bornToId": null,
    "deathDate": 1083,
    "gender": "F",
    "id": 2,
    "isMonarch": 0,
    "name": "Matilde of Flanders",
    "title": "Queen of England",
  },
  Object {
    "birthDate": 1056,
    "bornToId": 1,
    "deathDate": 1100,
    "gender": "M",
    "id": 3,
    "isMonarch": 1,
    "name": "William Rufus",
    "title": "King of England",
  },
]
`;

exports[`PersonModel update 1`] = `
Object {
  "birthDate": 1028,
  "bornToId": null,
  "deathDate": 1087,
  "gender": "M",
  "id": 1,
  "isMonarch": 1,
  "name": "Joe",
  "title": "King of England",
}
`;
